deployments/cloudfoundry/buildpack/bin/supply

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -euo pipefail
+
+BUILD_DIR=$1
+CACHE_DIR=$2
+DEPS_DIR=$3
+IDX=$4
+
+BUILDPACK_DIR="$(readlink -f ${BASH_SOURCE%/*})"
+TARGET_DIR="$BUILD_DIR/.otelcollector"
+
+OTEL_CONFIG="${OTEL_CONFIG:-otelconfig.yaml}"
+# Version must be >=0.47.0, the first release included the enabled Cloud Foundry receiver
+OTEL_VERSION="${OTEL_VERSION:-0.47.0}"
+# Cloud Foundry only supports Linux stacks currently, but if that changes this variable can be modified
+# for the proper stack.
+OS="${OS:-linux_amd64}"
+
+echo "-----> Installing Otel Collector ${OTEL_VERSION}"
+echo "       BUILD_DIR: $BUILD_DIR"
+echo "       CACHE_DIR: $CACHE_DIR"
+echo "       DEPS_DIR: $DEPS_DIR"
+echo "       BUILDPACK_INDEX: $IDX"
+echo "       BUILDPACK_DIR: $BUILDPACK_DIR"
+echo "       TARGET_DIR: $TARGET_DIR"
+
+mkdir -p $TARGET_DIR
+
+# Copy over default OpenTelemetry configuration file for sidecar configuration use
+cp "$BUILDPACK_DIR/../$OTEL_CONFIG" "$TARGET_DIR/"
+
+# File name for given release version
+OTELCOL="${OTELCOL:-otelcontribcol_${OS}-v${OTEL_VERSION}}"
+OTELCOL_TAR="$OTELCOL.tar.gz"
+CACHED_OTELCOL_TAR="$CACHE_DIR/$OTELCOL_TAR"
+
+if [[ -f "$CACHED_OTELCOL_TAR" ]]; then
+  echo "-----> Using cached Otel Collector install: $CACHED_OTELCOL_TAR"
+  cp $CACHED_OTELCOL_TAR $TARGET_DIR
+else
+  OTELCOL_DOWNLOAD_URL="${OTELCOL_DOWNLOAD_URL:-https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v${OTEL_VERSION}/otelcol-contrib_${OTEL_VERSION}_${OS}.tar.gz}"
+  echo "-----> Downloading OpenTelemetry Collector v$OTEL_VERSION ($OTELCOL_DOWNLOAD_URL)"
+  wget -O "$TARGET_DIR/$OTELCOL_TAR" $OTELCOL_DOWNLOAD_URL > /dev/null 2>&1
```

Comment:
Should we document in the readme that `wget` is a requirement?

Also, since you're redirecting all output to `/dev/null`, download failures would be silent.  Should there be a message printed for failures, or remove the redirection?

Reply:
I've added a note in the installation section of the README that wget is used.

There's a check directly after this that prints out an error message and stops the deployment if the download fails, but I've added the `--no-verbose` option to get rid of spam, but still print error messages if any occur.